### PR TITLE
Swift 4.2 Update

### PR DESCRIPTION
**Pull request by malinkas86**
* Change the source to support Swift 4.2

### CODE REVIEW

File: Marker.podspec [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
 Pod::Spec.new do |s|
   s.name             = "Marker"
-  s.version          = "1.1.0"
-  s.summary          = "A light wrapper around NSAttributedString."
```
Comment (DannyVancura) [inline Marker.podspec]: This should not change. `Swift 4.2 migration` is not the pod's summary.

**Review by DannyVancura — CHANGES_REQUESTED** [review]

File: Example/Marker-Example.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -446,12 +414,64 @@
 		271C85CF1ED4E3F000F8BBBB /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
```
Comment (ThibaultKlein) [inline Example/Marker-Example.xcodeproj/project.pbxproj]: What are all these additions for?

File: Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift [hunk 3]
```diff
@@ -29,11 +29,7 @@ public extension UITextView {
     ///   - textStyle: Text style object containing style information.
     func setMarkdownText(_ markdownText: String, using textStyle: TextStyle) {
         if let linkColor = textStyle.linkColor {
-            #if swift(>=4.0)
```
Comment (ThibaultKlein) [inline Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: Why did you remove this statement? Are we dropping Swift versions older than 4.0 in this PR as well? If that's the case, we should at least update the README for the minimum required versions of Swift and Xcode.

File: .travis.yml [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-osx_image: xcode9
+osx_image: xcode10
 podfile: Example/Podfile
 
 before_install:
     - gem install cocoapods --pre --no-rdoc --no-ri --no-document --quiet
 
 script:
-    - set -o pipefail && xcodebuild test -project Marker/Marker.xcodeproj -scheme Marker-iOS -destination 'name=iPhone 7,OS=11.0' | xcpretty
+    - set -o pipefail && xcodebuild test -project Marker/Marker.xcodeproj -scheme Marker-iOS -destination 'name=iPhone 8,OS=12.0' | xcpretty
```
Comment (ThibaultKlein) [inline .travis.yml:9]: You can specify the latest version of the OS instead of a specific version number, that might help the maintainability.

File: Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift (hunk 3 above)
Comment (malinkas86) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: It's the same for both old and new

File: .travis.yml (hunk 4 above)
Comment (malinkas86) [reply .travis.yml:9]: I couldn't find the syntax for it

File: Example/Marker-Example.xcodeproj/project.pbxproj (hunk 2 above)
Comment (malinkas86) [reply Example/Marker-Example.xcodeproj/project.pbxproj]: Upgrading the project to support Swift 4.2 may have included the setting.
Could take it out if not required

File: Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift (hunk 3 above)
Comment (DannyVancura) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: @ThibaultKlein @malinkas86 if I'm not mistaken, we are implicitly dropping older Swift versions with some of these changes already (e.g.: `UIApplication.LaunchOptionsKey`). Even 4.0 and 4.1 I think are dropped with this?

Comment (DannyVancura) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: Better example (since the above only affects the example project): `NSAttributedString.Key`

Comment (ThibaultKlein) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: Got it, as long as it's reflected in the README I'm fine with that. It should say at the top of the README what version of Swift is compatible with the current version.

File: .travis.yml (hunk 4 above)
Comment (ThibaultKlein) [reply .travis.yml:9]: I couldn't find it either, never mind then, I remember using a syntax like `OS=latest` but I can't remember on top of my head.

File: Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift (hunk 3 above)
Comment (malinkas86) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: Could this be marked as 'Resolved'?

**Review by DannyVancura — APPROVED** [review]

File: Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift (hunk 3 above)
Comment (DannyVancura) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: Hm the supported Swift version is currently not mentioned in the Readme at all as far as I see. I think it could be for sure if you make that addition. Conversely there's also a `spec.swift_version = '4.2'` we could add in the podspec instead and highlight the upgrade in the [release notes](https://github.com/prolificinteractive/Marker/releases) for this upgrade.

Comment (ThibaultKlein) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: @malinkas86 once the Swift supported version is documented in the spec file and README I'm good with this comment.

Comment (malinkas86) [reply Marker/Marker/Utility/Extensions/TextView/UITextViewExtension.swift:32]: Done.
